nuocd/nuodb_adminquery.py

```diff
@@ -137,23 +153,39 @@ def get_admin_conn(pid):
     try:
         # only interested in nuodb process on localhost, and don't
         # want to make nuoadmin rest call unless a new process is discovered.
-        _processes = subprocess.check_output(["pgrep", "^nuodb$"])
-        pids = _processes.split()
+
+        if not query_admin:
+            _processes = subprocess.check_output(["pgrep", "^nuodb$"])
+            pids = six.ensure_text(_processes).split()
+        else:
+            # if query admin, only do so if nuodb has not been found.
+            if len(running_local_processes) == 0:
+                pids = [ -1 ]
+            else:
+                pids = [ ]
```

Comment:
This query_admin stuff does not seem related to Python 2 vs Python 3. What are these changes and why do we need them for this PR?